### PR TITLE
Updates for python3, later versions of keras

### PR DESCRIPTION
**Pull request by osheroff**
👋 

I'm crazy happy to have found this repo.  I've been looking for an open source project that does reasonably accurate song segmentation for a couple of weeks now, and this is the first repo I've worked with that produces nice results, as well as having clean code (so rare in the world of academic research projects, go figure :) )

With the changes in this PR, I've got a model trained.  It's not quite as accurate as the figures you reported achieving but it's close.  I don't have the entirety of the SALAMI archive pulled in yet, but I plan to get more of it in (and perhaps even work on some of the annotation data).  Since it's 5 in the morning and I haven't slept well and I'm excited, I'm going to ask you some questions.

- are you still interested in this project or have a use for it?
- Is there an OSS license for it?
- Did you work with only the internet archive part of SALAMI in your training data?  
- Since it's generally a pain to get SALAMI downloaded, what would you say to releasing a pre-trained network in this repo?
- Did you ever experiment with the downbeat tracker as input?  (specifically, limiting segment transitions to a downbeat (1)) The tracker seemed fairly accurate at first blush, it'd be interesting to see if only using downbeats improved accuracy
- In some of the research papers they mentioned accentuating the "capitalized" segment markers in SALAMI.  Anecdotally, I've found that many (if not most) of the "lower case" segment markers are kinda just trash.  Did you do any experiments in this realm?

thanks for your time.  Boring PR stuff below:

most of the changes here deal with python not wanting to implicitly truncate floats to integers.  There's also some keras API changes.

There is also a fairly large change in track_segmentation.py, which is the call to np.normalize() on the features.

### CODE REVIEW

**Comment by mleimeister** [comment]
Hey, great to hear from you and really happy to hear the code is useful. It's interesting that there isn't more out there, especially I thought that there might be an official implementation eventually, but apparently not. I never got around to trying out the MSAF library (https://pythonhosted.org/msaf/index.html#) that contains a number of more classic segmentation algorithms. Have you had any experience with that?

I'm happy to merge your changes, and hope it was not too much of a hassle to get it working in a recent environment, as it hasn't been updated forever. One nice thing to add - I didn't do this initially either so it's actually more task for me - could be to specify which Python and library versions (esp. Keras) it is currently working with for you? I could then add this to the README.md.

Regarding your questions:

>     * are you still interested in this project or have a use for it?
I basically haven't continued to work on it since I moved away from music analysis to NLP some years ago. If you are interested to continue working on it, I would be happy to contribute a bit, though with very limited time resources.

>     * Is there an OSS license for it?
Not specified yet, but I'd be happy to add e.g. an MIT license, if that works for what you intend to do

>     * Did you work with only the internet archive part of SALAMI in your training data?
Yes, I didn't get very deep into exploring SALAMI at the time and the download script [here](https://github.com/DDMAL/SALAMI/blob/master/SALAMI_download.py) worked for that subset out of the box

>     * Since it's generally a pain to get SALAMI downloaded, what would you say to releasing a pre-trained network in this repo?
Yes, totally up for it :)
 
>     * Did you ever experiment with the downbeat tracker as input?  (specifically, limiting segment transitions to a downbeat (1)) The tracker seemed fairly accurate at first blush, it'd be interesting to see if only using downbeats improved accuracy
No, not in these experiments, but I agree that it could definitely improve performance quite a bit. From an analysis pov, I found it interesting though to see where the CNN might miss the downbeat and if that could be used to improve e.g. the features/architecture.

>     * In some of the research papers they mentioned accentuating the "capitalized" segment markers in SALAMI.  Anecdotally, I've found that many (if not most) of the "lower case" segment markers are kinda just trash.  Did you do any experiments in this realm?
Ah, right, that comes down to detecting segments at a more coarse level, right? I didn't try that, but sometimes the segment boundaries seemed indeed very close and sometimes even random

Hope that gives at least some first answers to your questions. If you like, we could also discuss these non-PR related things further via email?

Cheers...

File: Python/track_segmentation.py [hunk 1]
```diff
@@ -48,9 +48,10 @@ def extract_features(audio_file, beats_file):
     """
 
     t = pd.read_table(beats_file, header=None)
-    beat_times = t.ix[:, 0].values
+    beat_times = t.iloc[:, 0].values
 
     beat_mls = compute_beat_mls(filename=audio_file, beat_times=beat_times)
+    beat_mls /= np.max(beat_mls)
```
Comment (osheroff) [inline Python/track_segmentation.py:55]: @mleimeister this is the only line I added that wasn't just part of "upgrade land"; brefore this track_segmentation.py was giving sort of nonsensical results.  I'm not exactly sure why...

**Comment by osheroff** [comment]
sure, I'd love to discuss any / all.  I'm ben dot osheroff at gmail dot com

**Comment by mleimeister** [comment]
Hi @osheroff, sorry for not getting back before, the last weeks went by super fast. Thanks for the further commits, I'll add the license this week and merge in your changes.

**Comment by osheroff** [comment]
Ok.  I should probably clean up the work before merging, some of the changes in the branch are specific to my implementation
